unit-tests/func/rec-play/test-got-playback-frames.py

```diff
@@ -0,0 +1,245 @@
+    # will be True. If the record and playback failed it will be false
+    test.check( got_frames )
+except Exception:
+    test.unexpected_exception()
+finally: # we must remove all references to the file so we can use it again in the next test
+    if recorder:
+        recorder.pause()
+        recorder = None
+    if playback:
+        playback.pause()
+        playback = None
+    # if the sensor is already closed get_active_streams returns an empty list
+    if depth_sensor:
+        if depth_sensor.get_active_streams():
+            try:
+                depth_sensor.stop()
+            except RuntimeError as rte:
+                if str( rte ) != "stop_streaming() failed. UVC device is not streaming!":
+                    test.unexpected_exception()
+            except Exception:
+                test.unexpected_exception()
+            depth_sensor.close()
+        depth_sensor = None
+    if color_sensor:
+        if color_sensor.get_active_streams():
+            try:
+                color_sensor.stop()
+            except RuntimeError as rte:
+                if str(rte) != "stop_streaming() failed. UVC device is not streaming!":
+                    test.unexpected_exception()
+            except Exception:
+                test.unexpected_exception()
+            color_sensor.close()
+        color_sensor = None
+
+test.finish()
+
+#####################################################################################################
+test.start("Trying to record and playback using sensor interface with syncer")
+
+try:
+    sync = rs.syncer()
+    dev = test.find_first_device_or_exit()
+    recorder = rs.recorder(file_name, dev)
+    depth_sensor = dev.first_depth_sensor()
+    color_sensor = dev.first_color_sensor()
+
+    restart_profiles()
+
+    depth_sensor.open(dp)
+    depth_sensor.start(sync)
+    color_sensor.open(cp)
+    color_sensor.start(sync)
+
+    time.sleep(3)
+
+    recorder.pause()
+    recorder = None
+    color_sensor.stop()
+    color_sensor.close()
+    depth_sensor.stop()
+    depth_sensor.close()
+
+    ctx = rs.context()
+    playback = ctx.load_device(file_name)
+
+    depth_sensor = playback.first_depth_sensor()
+    color_sensor = playback.first_color_sensor()
+
+    restart_profiles()
+
+    depth_sensor.open(dp)
+    depth_sensor.start(sync)
+    color_sensor.open(cp)
+    color_sensor.start(sync)
+
+    # if the record-playback worked we will get frames, otherwise the next line will timeout and throw
+    sync.wait_for_frames()
+except Exception:
+    test.unexpected_exception()
+finally: # we must remove all references to the file so the temporary folder can be deleted
+    if recorder:
+        recorder.pause()
+        recorder = None
+    if playback:
+        playback.pause()
+        playback = None
+    # if the sensor is already closed get_active_streams returns an empty list
+    if depth_sensor:
+        if depth_sensor.get_active_streams():
```

Comment:
Looks like we need a `stop_pipeline()` functions, as well as a `stop_sensor()`

Reply:
I'm not sure I understand what you mean. There is pipeline.stop()

Reply:
I mean, there should be a function that takes care of all the `try` `except` calls and hides if from the code here

Reply:
where should I put them?